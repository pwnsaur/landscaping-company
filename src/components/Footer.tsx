import styled from 'styled-components';

const Footer = () => {
  return (
    <FooterContainer>
      <FooterText>Copyright 2023 Hackerman</FooterText>
    </FooterContainer>
  );
};

export default Footer;

const FooterContainer = styled.footer`
  display: flex;
  width: 100%;
  height: 15rem;
  justify-content: center;
  align-items: center;
  margin: auto 0 0;
  background-color: ${({ theme }) => theme.colors.darkGreen};
  /* z-index: 1; */
`;

const FooterText = styled.p`
  color: white;
`;
