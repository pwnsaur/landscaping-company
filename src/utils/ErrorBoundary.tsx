import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
