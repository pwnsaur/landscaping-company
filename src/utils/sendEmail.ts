import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
