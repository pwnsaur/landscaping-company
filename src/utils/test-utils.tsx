import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
