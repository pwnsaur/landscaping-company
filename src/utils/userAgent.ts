export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
