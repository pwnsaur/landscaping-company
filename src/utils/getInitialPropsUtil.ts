import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
