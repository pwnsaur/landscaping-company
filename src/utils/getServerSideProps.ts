import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
