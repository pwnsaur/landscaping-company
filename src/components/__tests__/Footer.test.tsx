import { render, screen } from '@/utils/test-utils';
import Footer from '@/components/Footer';

describe('Footer Component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Footer copyright', () => {
  test('text is correct', () => {
    render(<Footer />);
    const copyrightText = screen.getByText(/Copyright 2023 Hackerman/i);
    expect(copyrightText).toBeInTheDocument();
  });
});
