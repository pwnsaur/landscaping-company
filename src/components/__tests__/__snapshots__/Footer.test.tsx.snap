// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot 1`] = `
<DocumentFragment>
  .c1 {
  min-width: 9rem;
  height: 3rem;
  padding: 0 1rem;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 1rem;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #21412a;
  border: 2px solid #ededed;
  color: #ededed;
}

.c1:hover {
  background-color: #ededed;
  border-color: #21412a;
  color: #21412a;
}

.c0 {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 15rem;
  justify-content: center;
  align-items: center;
  margin: auto 0 0;
  padding-top: 5rem;
  background-color: #21412a;
}

.c2 {
  color: white;
  margin-bottom: 1rem;
  margin-top: auto;
}

<footer
    class="c0"
  >
    <a
      href="/contacts"
    >
      <button
        class="c1"
      >
        SazinƒÅties
      </button>
    </a>
    <p
      class="c2"
    >
      Copyright 2023 Hackerman
    </p>
  </footer>
</DocumentFragment>
`;
